doc/fips.md

```diff
@@ -0,0 +1,132 @@
+ * `LIBSPDM_SHA3_256_SUPPORT`, `LIBSPDM_SHA3_384_SUPPORT`, `LIBSPDM_SHA3_512_SUPPORT`
+
+Below algorithms will be disabled:
+ * `LIBSPDM_SM2_DSA_P256_SUPPORT`
+ * `LIBSPDM_SM2_KEY_EXCHANGE_P256_SUPPORT`
+ * `LIBSPDM_AEAD_CHACHA20_POLY1305_SUPPORT`
+ * `LIBSPDM_AEAD_SM4_128_GCM_SUPPORT`
+ * `LIBSPDM_SM3_256_SUPPORT`
+
+### FIPS approved algorithm
+
+If FIPS mode is enabled, then only FIPS-approved algorithms will be enabled, which is listed in [NIST.SP.800-140Cr1](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-140Cr1.pdf).
+
+### Key zeroization
+
+If a key is not used, then the variable to hold the key must be explictly zeroized. This is done in the libspdm.
+
+The private key for signing is managed by the [requester-asymsignlib](https://github.com/DMTF/libspdm/blob/main/include/hal/library/requester/reqasymsignlib.h) and [responder-asymlib](https://github.com/DMTF/libspdm/blob/main/include/hal/library/responder/asymsignlib.h). The library provider shall guarantee the key is zeroized after use.
+
+The pre-shared key (PSK) is managed by the [requester-psklib](https://github.com/DMTF/libspdm/blob/main/include/hal/library/requester/psklib.h) and [responder-psklib](https://github.com/DMTF/libspdm/blob/main/include/hal/library/responder/psklib.h). The library provider shall guarantee the key is zeroized after use.
+
+### module version API
+
+`libspdm_module_version()` in [spdm_common_lib](https://github.com/DMTF/libspdm/blob/main/include/library/spdm_common_lib.h) can return libspdm version information.
+
+### self-test API
+
+`libspdm_fips_run_selftest()` in [spdm_crypt_lib](https://github.com/DMTF/libspdm/blob/main/include/library/spdm_crypt_lib.h) can be used to run FIPS selftest, including
+ * `libspdm_fips_selftest_hmac_sha256()`
+ * `libspdm_fips_selftest_hmac_sha384()`
+ * `libspdm_fips_selftest_hmac_sha512()`
+ * `libspdm_fips_selftest_aes_gcm()`
+ * `libspdm_fips_selftest_rsa_ssa()`
+ * `libspdm_fips_selftest_hkdf()`
+ * `libspdm_fips_selftest_ecdh()`
+ * `libspdm_fips_selftest_sha256()`
+ * `libspdm_fips_selftest_sha384()`
+ * `libspdm_fips_selftest_sha512()`
+ * `libspdm_fips_selftest_sha3_256()`
+ * `libspdm_fips_selftest_sha3_384()`
+ * `libspdm_fips_selftest_sha3_512()`
+ * `libspdm_fips_selftest_ffdh()`
+ * `libspdm_fips_selftest_ecdsa()`
+ * `libspdm_fips_selftest_eddsa()`
+
+If any test failed, then `libspdm_fips_run_selftest()` will return false.
+
+`libspdm_fips_run_selftest()` requires `fips_selftest_context` parameter, which is initialized by `libspdm_get_fips_selftest_context_size()`, `libspdm_init_fips_selftest_context()` in [spdm_common_lib](https://github.com/DMTF/libspdm/blob/main/include/library/spdm_common_lib.h).
+
+The expected step is as follows:
+1) The integrator invokes `libspdm_get_fips_selftest_context_size()` and `libspdm_init_fips_selftest_context()` to create the FIPS selftest context.
+2) The integrator invokes `libspdm_fips_run_selftest()` to trigger self-test.
```

Comment:
As commented above, libspdm_fips_run_selftest() should be triggered by libspdm itself.  Is the following sample code part of libspdm?

Reply:
No.
We added it before, but it seems cause lots of confusing. Finally, we decide to let integrator invoke libspdm_fips_run_selftest() in 3.0.0.

We can revisit later if there is need.